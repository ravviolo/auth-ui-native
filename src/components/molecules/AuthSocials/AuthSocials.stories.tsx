import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { StoryScreen } from 'storybook/StoryScreen';

import { AuthSocials } from './AuthSocials';

storiesOf('AuthSocials', module)
  .addDecorator((getStory) => <StoryScreen>{getStory()}</StoryScreen>)
  .add('Default', () => (
    <AuthSocials
      testID={text('testID', 'auth-socials-test-id')}
      onPressFacebook={action('onPressFacebook')}
      onPressGoogle={action('onPressGoogle')}
      onPressLinkedIn={action('onPressLinkedIn')}
    />
  ));
