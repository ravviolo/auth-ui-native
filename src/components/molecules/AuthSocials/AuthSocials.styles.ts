import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'space-around',
    paddingHorizontal: 55,
  },
});
