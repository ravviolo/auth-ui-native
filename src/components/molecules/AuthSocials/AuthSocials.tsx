import { Flex, IconButton } from 'components/atoms';
import { StyleSheet } from 'react-native';

interface Props {
  testID: string;
  onPressGoogle: () => void;
  onPressFacebook: () => void;
  onPressLinkedIn: () => void;
}

interface AuthSocial {
  name: string;
  color: string;
  icon: string;
  onPressHandler: () => void;
}

export const AuthSocials = ({ onPressFacebook, onPressGoogle, onPressLinkedIn, testID }: Props) => {
  const authSocials: AuthSocial[] = [
    {
      name: 'Google',
      color: '#DB4437',
      icon: 'G',
      onPressHandler: onPressGoogle,
    },
    {
      name: 'Facebook',
      color: '#4267B2',
      icon: 'f',
      onPressHandler: onPressFacebook,
    },
    {
      name: 'LinkedIn',
      color: '#0077B5',
      icon: 'in',
      onPressHandler: onPressLinkedIn,
    },
  ];
  return (
    <Flex direction="row" style={styles.container} testID={testID}>
      {authSocials.map(({ onPressHandler, color, icon, name }) => (
        <IconButton
          key={`icon-btn-${name}`}
          color={color}
          icon={icon}
          testID={`icon-btn-${name}-test-id`}
          onPress={onPressHandler}
        />
      ))}
    </Flex>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'space-around',
    paddingHorizontal: 55,
  },
});
