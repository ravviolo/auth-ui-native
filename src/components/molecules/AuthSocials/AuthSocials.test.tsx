import { fireEvent, render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { AuthSocials } from './AuthSocials';

import type { Props } from './AuthSocials';

describe('AuthSocials', () => {
  const mockOnPressFacebook = jest.fn<Props['onPressFacebook'], []>();
  const mockOnPressGoogle = jest.fn<Props['onPressGoogle'], []>();
  const mockOnPressLinkedIn = jest.fn<Props['onPressLinkedIn'], []>();

  const props: Props = {
    onPressFacebook: mockOnPressFacebook,
    onPressGoogle: mockOnPressGoogle,
    onPressLinkedIn: mockOnPressLinkedIn,
  };

  it('should render 3 icon buttons', () => {
    render(<AuthSocials {...props} />);

    const authSocialsIcons = screen.getAllByTestId(/icon-btn-(facebook|google|linkedin)-test-id/i);

    expect(authSocialsIcons).toHaveLength(3);
  });

  it('should handle click events on Facebook icon - run handler once, not trigger other handlers', () => {
    render(<AuthSocials {...props} />);

    const facebookIconBtn = screen.getByTestId(/icon-btn-facebook-test-id/i);

    fireEvent.press(facebookIconBtn);

    expect(mockOnPressFacebook).toHaveBeenCalledTimes(1);
    expect(mockOnPressGoogle).not.toBeCalled();
    expect(mockOnPressLinkedIn).not.toBeCalled();
  });

  it('should handle click events on Google icon - run handler once, not trigger other handlers', () => {
    render(<AuthSocials {...props} />);

    const googleIconBtn = screen.getByTestId(/icon-btn-google-test-id/i);

    fireEvent.press(googleIconBtn);

    expect(mockOnPressGoogle).toHaveBeenCalledTimes(1);
    expect(mockOnPressFacebook).not.toBeCalled();
    expect(mockOnPressLinkedIn).not.toBeCalled();
  });

  it('should handle click events on LinkedIn icon - run handler once, not trigger other handlers', () => {
    render(<AuthSocials {...props} />);

    const linkedInIconBtn = screen.getByTestId(/icon-btn-linkedin-test-id/i);

    fireEvent.press(linkedInIconBtn);

    expect(mockOnPressLinkedIn).toHaveBeenCalledTimes(1);
    expect(mockOnPressFacebook).not.toBeCalled();
    expect(mockOnPressGoogle).not.toBeCalled();
  });

  it('should match snapshot', () => {
    const tree = testRenderer.create(<AuthSocials {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
