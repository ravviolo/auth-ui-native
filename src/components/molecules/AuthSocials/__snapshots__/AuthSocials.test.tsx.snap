// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthSocials should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "space-around",
        "paddingHorizontal": 55,
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={20}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        Object {
          "alignItems": "center",
          "borderRadius": 100,
          "borderWidth": 2,
          "height": 50,
          "justifyContent": "center",
          "padding": 8,
          "width": 50,
        },
        Object {
          "borderColor": "#DB4437",
        },
      ]
    }
    testID="icon-btn-Google-test-id"
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
          },
          Object {
            "color": "#DB4437",
          },
        ]
      }
    >
      G
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={20}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        Object {
          "alignItems": "center",
          "borderRadius": 100,
          "borderWidth": 2,
          "height": 50,
          "justifyContent": "center",
          "padding": 8,
          "width": 50,
        },
        Object {
          "borderColor": "#4267B2",
        },
      ]
    }
    testID="icon-btn-Facebook-test-id"
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
          },
          Object {
            "color": "#4267B2",
          },
        ]
      }
    >
      f
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={20}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        Object {
          "alignItems": "center",
          "borderRadius": 100,
          "borderWidth": 2,
          "height": 50,
          "justifyContent": "center",
          "padding": 8,
          "width": 50,
        },
        Object {
          "borderColor": "#0077B5",
        },
      ]
    }
    testID="icon-btn-LinkedIn-test-id"
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
          },
          Object {
            "color": "#0077B5",
          },
        ]
      }
    >
      in
    </Text>
  </View>
</View>
`;
